### PR TITLE
Make interface for routing table (#2046)

### PR DESCRIPTION
**Pull request by sindydwns**
Issue address is #2046

### CODE REVIEW

**Review by longfin — COMMENTED** [review]
How about removing redudant comments on the implmentation side(i.e., `RoutingTable`) with [`<inheritdoc/>`](https://docs.microsoft.com/ko-kr/dotnet/csharp/language-reference/xmldoc/recommended-tags#inheritdoc) in this PR too?

**Review by dahlia — DISMISSED** [review]

**Review by dahlia — DISMISSED** [review]

**Review by OnedgeLee — DISMISSED** [review]

**Comment by longfin** [comment]
@sindydwns latest change looks good to me. but sadly conflicting occured after merging #2201. 😢 


could you check it? 🙏

**Review by OnedgeLee — APPROVED** [review]

File: CHANGES.md
Comment (tkiapril) [inline CHANGES.md:91]:
```suggestion
 -  Added `IRoutingTable` interface.  [[#2046], [#2229]]
```

**Review by tkiapril — APPROVED** [review]
Please rebase and resolve merge conflict in CHANGES.md 🙏

**Review by longfin — DISMISSED** [review]

**Review by longfin — APPROVED** [review]

**Review by riemannulus — APPROVED** [review]

**Review by OnedgeLee — DISMISSED** [review]

**Comment by OnedgeLee** [comment]
Seems like rebase on maintainer side would be better, since main branch is frequently updated recently

**Review by tkiapril — APPROVED** [review]

File: CHANGES.md
Comment (tkiapril) [inline CHANGES.md:157]: Please move your changelog to To be released section.

File: CHANGES.md
Comment (tkiapril) [inline CHANGES.md:211]: Please move your changelog to To be released section.

File: CHANGES.md
Comment (tkiapril) [inline CHANGES.md:221]: Please move your changelog to To be released section.

**Review by tkiapril — CHANGES_REQUESTED** [review]

**Review by tkiapril — DISMISSED** [review]

File: CHANGES.md
Comment (sindydwns) [reply CHANGES.md]: Thanks for letting me know. i move v0.41.0 to v0.42.0.

**Review by tkiapril — DISMISSED** [review]

File: CHANGES.md
Comment (limebell) [inline CHANGES.md:23]:
```suggestion
 -  Added `IRoutingTable` interface.  [[#2046], [#2229]]
 -  `RoutingTable` became to implement `IRoutingTable` interface.  [[#2046], [#2229]]
```

File: CHANGES.md
Comment (limebell) [inline CHANGES.md:24]:
```suggestion
 -  `RoutingTable` became to implement `IRoutingTable` interface.
    [[#2046], [#2229]]
```

**Review by tkiapril — APPROVED** [review]

**Pull request merged**
